interfaces/builtin/avahi_observe.go

```diff
@@ -430,10 +429,12 @@ func (iface *avahiObserveInterface) StaticInfo() interfaces.StaticInfo {
 func (iface *avahiObserveInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	old := "###SLOT_SECURITY_TAGS###"
 	var new string
-	if release.OnClassic {
-		// If we're running on classic Avahi will be part
-		// of the OS snap and will run unconfined.
-		new = "unconfined"
+	// If we're running on classic, Avahi may be installed either as a snap of
+	// as part of the OS. If it is part of the OS, it will not have a security
+	// label like it would when installed as a snap.
+	if implicitSystemConnectedSlot(slot) {
+		// avahi from the OS is typically unconfined but known to sometimes be confined
+		new = "{unconfined,/usr/sbin/avahi-daemon}"
```

Comment:
Please change this to:
```
		new = "\"{unconfined,/usr/sbin/avahi-daemon}\""
```

Reply:
Thank you @jdstrand 
This is now changed and tested. Now it works when I use avahi-client and connect its plug to implicit slot